File: algorithms/reverse_string_in_place/reverseStringInPlace.ts
const reverseString = (str: string): string => {
  let start = 0;
  let end = str.length - 1;

  let chars = str.split("");

  while (start < end) {
    let temp = chars[start];
    chars[start] = chars[end];
    chars[end] = temp;

    start++;
    end--;
  }

  return chars.join("");
};

export default reverseString;

File: algorithms/reverse_string_in_place/reverse_string_in_place.py
def reverse_chars(chars):
    # Initialize two pointers to track the start and end of the list
    start = 0
    end = len(chars)-1

    while start < end:
        chars[start], chars[end] = chars[end], chars[start]
        start += 1
        end -= 1

    return chars
